services/src/main/java/org/fao/geonet/api/records/editing/InspireValidatorUtils.java

```diff
@@ -509,4 +526,54 @@ public String submitFile(String serviceEndpoint, InputStream record, String test
             // client.close();
         }
     }
+
+    /**
+     *  Executes the HttpUriRequest
+        * @param request
+        * @return response of execute method
+        * @throws IOException
+     */
+    private ClientHttpResponse execute(HttpUriRequest request) throws IOException {
+
+        boolean useProxy = settingManager.getValueAsBool(Settings.SYSTEM_PROXY_USE, false);
+        if (useProxy) {
+            String proxyHost = settingManager.getValue(Settings.SYSTEM_PROXY_HOST);
+            String proxyPort = settingManager.getValue(Settings.SYSTEM_PROXY_PORT);
+            String username = settingManager.getValue(Settings.SYSTEM_PROXY_USERNAME);
+            String password = settingManager.getValue(Settings.SYSTEM_PROXY_PASSWORD);
+
+            final Function<HttpClientBuilder, Void> proxyConfiguration = new Function<HttpClientBuilder, Void>() {
+                @Nullable
+                @Override
+                public Void apply(@Nonnull HttpClientBuilder input) {
+
+                    HttpHost proxy = null;
+                    if(StringUtils.isNotBlank(proxyPort)) {
+                        proxy = new HttpHost(proxyHost, Integer.valueOf(proxyPort));
+                    } else {
+                        proxy = new HttpHost(proxyHost);
+                    }
+
+                    input.setProxy(proxy);
+
+                    boolean isAuthenticationEnabled = StringUtils.isNotBlank(username);
+
+                    if (isAuthenticationEnabled) {
+                        Credentials credentials = new UsernamePasswordCredentials(username, password);
+                        AuthScope authScope = new AuthScope(proxyHost, Integer.parseInt(StringUtils.isNotBlank(proxyPort)?proxyPort:"80"));
+                        final BasicCredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+                        credentialsProvider.setCredentials(authScope, credentials);
+                        input.setDefaultCredentialsProvider(credentialsProvider);
+                    }
+
+                    return null;
+                }
+            };
+
+            return requestFactory.execute(request, proxyConfiguration);
+        } else {
+            return requestFactory.execute(request);
+        }
```

Comment:
This can be replaced by `org.fao.geonet.lib.netLib.setupProxy(ServiceContext context, HttpClientBuilder client, String requestHost)`
https://github.com/geonetwork/core-geonetwork/blob/72384b4a2601d01a3c4fc9dd23eedf6fcbb4902d/core/src/main/java/org/fao/geonet/lib/NetLib.java#L110

like in https://github.com/geonetwork/core-geonetwork/blob/72384b4a2601d01a3c4fc9dd23eedf6fcbb4902d/harvesters/src/main/java/org/fao/geonet/kernel/harvest/harvester/webdav/WebDavRetriever.java#L89-L90

Reply:
Done, thanks for the suggestion